### PR TITLE
Fix possible typos in reinterpreting effects doc

### PR DESCRIPTION
**Pull request by shinzui**

### CODE REVIEW

File: docs/reinterpreting_effects.md [hunk 1]
```diff
@@ -234,7 +234,7 @@ instance (MonadIO m, Algebra sig m) => Algebra (Http :+: sig) (MockHttpClient m)
     L (SendRequest req) -> do
       responder <- MockHttpClient ask
       (<$ ctx) <$> liftIO (responder req)
-    R other -> MockHttpClient (alg (runMockHttpClient . hdl) other ctx)
+    R other -> MockHttpClient (alg (runMockHttpClient . hdl) (R other) ctx)
```
Comment (ocharles) [inline docs/reinterpreting_effects.md:237]: I usually use `inj` here - is there any preference to what we should do?

Comment (shinzui) [reply docs/reinterpreting_effects.md]: I updated the doc to use `inj`. Is there any advantage of using it?

Comment (ocharles) [reply docs/reinterpreting_effects.md]: Sorry, I wasn't trying to /suggest/ `inj`, just that I use it. I don't know which is better!

Comment (shinzui) [reply docs/reinterpreting_effects.md]: I'm new to `fused-effects`, so I wouldn't know. But having to import `inj` from `Control.Effect.Sum` indicates, to me at least, that it's not meant to be used directly. Maybe @patrickt or  @robrix can weigh in.

Comment (robrix) [reply docs/reinterpreting_effects.md]: Generally I use `R`, but `inj` is fine too. As @shinzui notes, `inj` requires an extra import, so I’d probably use `R` in the docs.

File: docs/reinterpreting_effects.md [hunk 2]
```diff
@@ -181,7 +182,7 @@ handlePrint r =
     Left invalidContentTypeError -> print invalidContentTypeError
     Right ok -> case ok of
       Left jsonParseError -> print jsonParseError
-      Right facts -> traverse (putStrLn . catFact) facts
+      Right facts -> mapM_ (putStrLn . catFact) facts
```
Comment (robrix) [inline docs/reinterpreting_effects.md:185]: Good catch 👍🏻

**Review by robrix — APPROVED** [review]
Nice, thank you! 👍🏻 I’m ok with leaving this using `inj`, or you can revert it to use `R`, at your option.

File: docs/reinterpreting_effects.md (hunk 1 above)
Comment (ocharles) [reply docs/reinterpreting_effects.md]: Cool, I'll probably switch to using `R` too then!

File: docs/reinterpreting_effects.md (hunk 2 above)
Comment (jkachmar) [reply docs/reinterpreting_effects.md:185]: Super tiny nitpick, but should be `mapM_` (in the spirit of this all being `Monad`ic anyway) or should it be `traverse_` (in the spirit of the original `traverse` call)?

Comment (shinzui) [reply docs/reinterpreting_effects.md]: Switched to `traverse_`

**Comment by shinzui** [comment]
I reverted back to using `R` and addressed the `mapM` comment. I can squash the commits if everything else looks fine.

File: docs/reinterpreting_effects.md (hunk 2 above)
Comment (robrix) [reply docs/reinterpreting_effects.md]: I’m good either way 👍🏻

**Comment by robrix** [comment]
@shinzui I (very strongly) prefer non-squashed history. Thanks very much for the great PR!

**Pull request merged**